src/engine/layoutstore.ts

```diff
@@ -28,20 +28,24 @@ class LayoutStoreEntry {
   private layouts: { [key: string]: ILayout };
   private previousID: string;
 
-  constructor(output_name: string) {
+  constructor(output_name: string, desktop_name?: string) {
```

Comment:
desktop_name always exists

Reply:
Depended on "Use separate layouts for each Desktop" setting